src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java

```diff
@@ -555,6 +562,53 @@ private void insertCreateSiteAuditLog(String siteId, String siteName, Map<String
         auditServiceInternal.insertAuditLog(auditLog);
     }
 
+    private void processCreatedFiles(String siteId, Map<String, String> createdFiles, String creator,
+                                     ZonedDateTime now, String lastCommitId) {
+        long startProcessCreatedFilesMark = System.currentTimeMillis();
+        StringBuilder sb = new StringBuilder();
+        int counter = 0;
+        int batchCounter = 0;
+        long startBatchMark = System.currentTimeMillis();
+        for (String path : createdFiles.keySet()) {
+            if (counter++ >= batchSize) {
+                studioDBScriptRunner.execute(sb.toString());
+                logger.debug("Process created files batch " + ++batchCounter + " in " +
+                        (System.currentTimeMillis() - startBatchMark) + " milliseconds");
+                sb = new StringBuilder();
+                counter = 0;
+                startBatchMark = System.currentTimeMillis();
+            }
+            sb.append(insertItemStateRow(siteId, path)).append("\n\n");
+            sb.append(insertItemMetadataRow(siteId, path, creator, now, lastCommitId)).append("\n\n");
+
+            boolean isXml = path.endsWith(DmConstants.XML_PATTERN);
```

Comment:
Are these hardcoded like this? Shouldn't this be a method that's used across Studio to decide if dep-calc is required?